clang/lib/Lex/HeaderSearch.cpp

```diff
@@ -142,6 +142,21 @@ std::vector<bool> HeaderSearch::computeUserEntryUsage() const {
   return UserEntryUsage;
 }
 
+std::vector<bool> HeaderSearch::computeVFSUsage() const {
+  std::vector<bool> VFSUsage;
+  llvm::vfs::FileSystem &RootFS = FileMgr.getVirtualFileSystem();
+  // TODO: This only works if the `RedirectingFileSystem`s were all created by
```

Comment:
Could all VFS created via `createVFSFromOverlayFiles` be marked with a flag and then assert here if the bit is not set? Or does it not matter and we only use the optimization potential?